### PR TITLE
factorial and fibonacci ruby programs

### PR DESCRIPTION
**Pull request by preethi900**
**Changes:**
* Initial ruby programs in the ruby folder

### CODE REVIEW

**Comment by Srinivas11789** [comment]
@preethi900 please add comments to your code.

**Comment by preethi900** [comment]
Added comments for the factorial leap year and string reverse programs

File: Ruby/factorial.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  1
+ else
+#Condition to find the factorial of a program
+  if n> 0
+   n * fact(n-1)
+   
+  else 
+   puts "no factorial for -ve number"
+  end
+end
```
Comment (Srinivas11789) [inline Ruby/factorial.rb]: indent is improper here

File: Ruby/fibonacci.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        a = b
+        # Add up previous two numbers in sequence.
+        b = temp + b
+    end
+
+    return a
+end
+
+# Write first 15 Fibonacci numbers in sequence.
+15.times do |n|
```
Comment (Srinivas11789) [inline Ruby/fibonacci.rb]: can we not hardcode this and get input from user?

File: Ruby/leap.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
+#Program to find if the year is a leap year or not
+
+#Get the starting year from user
+puts "Enter starting year:"
+
+
+        starting_year = gets.chomp.to_i
```
Comment (Srinivas11789) [inline Ruby/leap.rb]: indent improper again

File: Ruby/reverse.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+#Returning the reverse string
+  reverse
+end
+#Getting the string from console
+puts "Enter the string"
+string = gets
+#call the Reverse function to reverse the string
+reversedstring = reverse(string)
+#print the reversed string
+puts reversedstring
```
Comment (Srinivas11789) [inline Ruby/reverse.rb]: this code looks good.

File: Ruby/factorial.rb (hunk 1 above)
Comment (preethi900) [reply Ruby/factorial.rb]: Comment addressed

File: Ruby/fibonacci.rb (hunk 2 above)
Comment (preethi900) [reply Ruby/fibonacci.rb]: comment addressed

File: Ruby/leap.rb (hunk 3 above)
Comment (preethi900) [reply Ruby/leap.rb]: Comment addressed

**Comment by preethi900** [comment]
comments addressed

**Comment by Srinivas11789** [comment]
@preethi900 Thanks for your contribution

**Pull request merged**